### PR TITLE
[bitnami/elasticsearch] Istio showing Passthrough traffic among Elastic Pods

### LINKED ISSUE #2539: [bitnami/elasticsearch] Istio showing Passthrough traffic among Elastic Pods
**Which chart**:
Elasticsearch version 12.0.1

**Describe the bug**
Kiali dashboard shows that there is Passthrough traffic on port 9200 and 9300 anong Elastic Pods, which should be instead monitored by Istio and reported as part of the service mesh.

**To Reproduce**
1. Deploy Istio 1.5.2
1. Enable auto sidecar injection in the namespace
1. Deploy the chart with values production
1. Open Kiali and check the mesh graph, it shows Passthrough traffic among Elastic nodes

**Expected behavior**
No Passthrough traffic should happen among nodes

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}

```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:50:46Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}

```

### PR DESCRIPTION
**Pull request by irizzant**
**Description of the change**

Removes the Passthrough cluster among Elastic pods

**Benefits**

Enhance Istio compatibility

**Possible drawbacks**

None

**Applicable issues**

  - fixes #2539

**Additional information**

**Checklist** 
- [ X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ X] Variables are documented in the README.md
- [ X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [ X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/elasticsearch/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: elasticsearch
-version: 12.0.1
+version: 12.0.2
```
Comment (juan131) [inline bitnami/elasticsearch/Chart.yaml:3]: You're adding new features (e.g. podlabels). Therefore, you'll have to bump the minor version

File: bitnami/elasticsearch/templates/coordinating-svc.yaml [hunk 2]
```diff
@@ -20,5 +20,7 @@ spec:
       {{- else if eq .Values.coordinating.service.type "ClusterIP" }}
       nodePort: null
       {{- end }}
+    - name: tcp-transport
+      port: 9300
```
Comment (juan131) [inline bitnami/elasticsearch/templates/coordinating-svc.yaml:24]: Why is this necessary? This service is intended for "coordinating-only" nodes. Why do you need to access this port on this nodes?

File: bitnami/elasticsearch/templates/data-statefulset.yaml [hunk 3]
```diff
@@ -107,6 +110,8 @@ spec:
             - name: ELASTICSEARCH_NODE_TYPE
               value: "data"
           ports:
+            - name: http
+              containerPort: 9200
```
Comment (juan131) [inline bitnami/elasticsearch/templates/data-statefulset.yaml:114]: Why exposing this port on data nodes? These nodes shouldn't be accessed on this port by anyone.

File: bitnami/elasticsearch/templates/data-svc.yaml [hunk 4]
```diff
@@ -8,6 +8,9 @@ spec:
   type: ClusterIP
   publishNotReadyAddresses: true
   ports:
+    - name: http
+      port: 9200
+      targetPort: http
```
Comment (juan131) [inline bitnami/elasticsearch/templates/data-svc.yaml:13]: idem

File: bitnami/elasticsearch/templates/coordinating-svc.yaml (hunk 2 above)
Comment (irizzant) [reply bitnami/elasticsearch/templates/coordinating-svc.yaml:24]: This Service is reached on port 9300 from the other nodes, which use this port for different reasons, for example for zen discovery and cluster assembling or when an HTTP call is made to one node and a different one is holding the data.
This is needed to monitor this kind of traffic

File: bitnami/elasticsearch/templates/data-statefulset.yaml (hunk 3 above)
Comment (irizzant) [reply bitnami/elasticsearch/templates/data-statefulset.yaml:114]: Actually this port is already used by readiness and liveness [probes](https://github.com/irizzant/charts-1/blob/0d29428887f122512627ca744fce8a726c9f2254/bitnami/elasticsearch/templates/data-statefulset.yaml#L126), so there is traffic on it from other Pods (kubelet in this case).
Moreover having this port open gives the possibility to use REST API with these nodes too whenever needed.

File: bitnami/elasticsearch/templates/data-svc.yaml (hunk 4 above)
Comment (irizzant) [reply bitnami/elasticsearch/templates/data-svc.yaml:13]: Actually this port is already used by readiness and liveness [probes](https://github.com/irizzant/charts-1/blob/0d29428887f122512627ca744fce8a726c9f2254/bitnami/elasticsearch/templates/data-statefulset.yaml#L126), so there is traffic on it from other Pods (kubelet in this case).
Moreover having this port open gives the possibility to use REST API with these nodes too whenever needed.

File: bitnami/elasticsearch/Chart.yaml (hunk 1 above)
Comment (irizzant) [reply bitnami/elasticsearch/Chart.yaml:3]: Chart version bumped

File: bitnami/elasticsearch/templates/coordinating-svc.yaml (hunk 2 above)
Comment (juan131) [reply bitnami/elasticsearch/templates/coordinating-svc.yaml:24]: Agree on this! We can expose 9300 on coordinating-only nodes

File: bitnami/elasticsearch/templates/data-statefulset.yaml (hunk 3 above)
Comment (juan131) [reply bitnami/elasticsearch/templates/data-statefulset.yaml:114]: Okay with adding the port for the container since it's used for probes

File: bitnami/elasticsearch/templates/data-svc.yaml (hunk 4 above)
Comment (juan131) [reply bitnami/elasticsearch/templates/data-svc.yaml:13]: The probes are run locally (localhost), therefore the traffic never gets routed by the associated service.

On the other hand, why would you like to access the REST API from the outside on other node but the coordinating-only one?

I'm okay with adding the port to the pod definition (data-statefulset.yaml) but I don't see any advantage on adding it to the service.

Comment (irizzant) [reply bitnami/elasticsearch/templates/data-svc.yaml:13]: >The probes are run locally (localhost), therefore the traffic never gets routed by the associated service.

The probe traffic is generated by Kubelet, not inside the Pod itself even if the URL is a localhost one.

File: bitnami/elasticsearch/templates/coordinating-svc.yaml (hunk 2 above)
Comment (irizzant) [reply bitnami/elasticsearch/templates/coordinating-svc.yaml:24]: The comment above applies to all nodes, since they use 9300 to assemble the cluster and also to periodically check on each others presence.

**Comment by juan131** [comment]
I'll have to relaunch the jobs due to the conflicts. Doing so
